errai-security/errai-security-client/src/main/java/org/jboss/errai/security/client/local/context/SecurityContextImpl.java

```diff
@@ -111,6 +112,7 @@ public HTMLElement getElement() {
 
   @PostConstruct
   private void setup() {
+    userCache = Factory.maybeUnwrapProxy(userCache);
```

Comment:
So why is it a problem here that the `userCache` is proxied? (If that is a case, a cleaner fix is to change the cache implementation from `@ApplicationScoped` to `@Singleton`).